CRM/Core/BAO/CustomValueTable.php

```diff
@@ -641,6 +641,10 @@ public static function setValues(&$params) {
           $cvParam['file_id'] = $fieldValue['value'];
         }
 
+        if ($cvParam['type'] == 'Memo' && $cvParam['value']) {
+          $cvParam['value'] = html_entity_decode($cvParam['value']);
+        }
+
         if (!array_key_exists($dao->table_name, $cvParams)) {
```

Comment:
I'm not a fan of special casing field handling based on name, even though this is the existing pattern CiviCRM uses. It complicates what we need to know in order to predict behaviour, and the logic is scattered through multiple locations in CiviCRM.

If as CRM-11532 says CiviCRM's API should now handle input and output without escaping, then the correct fix here seems to be eliminating whatever is encoding that input before storage.